enzy_htp/_interface/rosetta_interface.py

```diff
@@ -38,7 +46,1113 @@
+
+        Args:
+            cst_gen (RosettaScriptsElement): The RosettaScriptsElement representing the constraint generator.
+
+        Returns:
+            RosettaScriptsProtocol: The current instance of the protocol with the added constraint generator.
+        """
+        return self.add_element('CONSTRAINT_GENERATORS', cst_gen)
+
+    @dispatch
+    def add_constraint_generator(self, cg_name: str, **kwargs) -> RosettaScriptsProtocol:
+        """Creates and adds a constraint generator using a name and additional parameters.
+
+        Args:
+            cg_name (str): The name of the constraint generator.
+            **kwargs: Additional parameters to configure the constraint generator.
+
+        Returns:
+            RosettaScriptsProtocol: The current instance of the protocol with the added constraint generator.
+        """
+        return self.add_element('CONSTRAINT_GENERATORS',
+            RosettaScriptsElement(cg_name, **kwargs)
+        )
+
+    @dispatch
+    def add_scoring_grid(self, scoring_grid: RosettaScriptsElement) -> RosettaScriptsProtocol:
+        """Adds a scoring grid element to the 'SCORINGGRIDS' section.
+
+        Args:
+            scoring_grid (RosettaScriptsElement): The RosettaScriptsElement representing the scoring grid.
+
+        Returns:
+            RosettaScriptsProtocol: The current instance of the protocol with the added scoring grid.
+        """
+        return self.add_element("SCORINGGRIDS", scoring_grid)
+
+    @dispatch
+    def add_scoring_grid(self, **kwargs) -> RosettaScriptsProtocol:
+        """Creates and adds a scoring grid element to the 'SCORINGGRIDS' section using specified parameters.
+
+        This method is designed to handle the creation of a scoring grid element with the provided keyword arguments,
+        which will be passed to the RosettaScriptsElement constructor. The name of the scoring grid is set as an
+        empty string by default.
+
+        Args:
+            **kwargs: Additional parameters to configure the scoring grid element.
+
+        Returns:
+            RosettaScriptsProtocol: The current instance of the protocol with the added scoring grid element.
+        """
+        return self.add_element("SCORINGGRIDS",
+            RosettaScriptsElement('', **kwargs)
+        )
+
+    @dispatch
+    def add_task_operations(self, task_op: RosettaScriptsElement) -> RosettaScriptsProtocol:
+        """Adds a task operations element to the 'TASKOPERATIONS' section.
+
+        This method allows for the addition of a pre-existing RosettaScriptsElement representing task operations
+        directly to the TASKOPERATIONS section of the protocol.
+
+        Args:
+            task_op (RosettaScriptsElement): The RosettaScriptsElement representing the task operations to be added.
+
+        Returns:
+            RosettaScriptsProtocol: The current instance of the protocol with the added task operations element.
```

Comment:
This block of methods from line 708 to 855 seems to be a duplication of the methods from line 423 to 598. Please remove the duplicated block.